### PR TITLE
SHARD-913 Added ENV for config and Docker image building workflow #74

### PR DESCRIPTION
**Pull request by arhamj**

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  display-image-name:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Display IMAGE_NAME
+        run: echo "IMAGE_NAME is ${{ env.IMAGE_NAME }}"
```
Comment (aniketdivekar) [inline .github/workflows/docker.yml:21]: Do we have tag w/ image name? or we are just overwriting one image?

**Review by aniketdivekar — DISMISSED** [review]

File: .github/workflows/docker.yml (hunk 1 above)
Comment (arhamj) [reply .github/workflows/docker.yml:21]: Tag is defined in the section`Build and push Docker image`. The convention after discussion with devops is to use {branch-short_commit_hash}

**Review by aniketdivekar — DISMISSED** [review]

**Comment by chrypnotoad** [comment]
@arhamj @aniketdivekar This patch no longer applies because of the config.ts changes
`cache-test-settings.patch`

**Comment by arhamj** [comment]
> @arhamj @aniketdivekar This patch no longer applies because of the config.ts changes `cache-test-settings.patch`

@chrypnotoad I have fixed the patches for now. Ideally, I would have wanted to delete these patches in favour of using env variables but, I don't have a solution to env management for local just yet. Thanks for catching it early. I've also added a `workflow_dispatcher` trigger as per the suggestion but since this is a public repo I'm not sure if anyone will be able to trigger this or if it's only members who can do it etc.

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+on:
+  push:
+    branches: ['dev']
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag for the Docker image'
+        required: false
+        default: 'latest' # Default tag if none is provided
```
Comment (chrypnotoad) [inline .github/workflows/docker.yml]: this input isn't being used when the tag is set

Comment (arhamj) [reply .github/workflows/docker.yml]: Updated.

**Review by chrypnotoad — APPROVED** [review]

**Pull request merged**